src/server/tran_server.cpp

```diff
@@ -145,7 +145,7 @@ tran_server::push_request (tran_to_page_request reqid, std::string &&payload)
   std::shared_lock<std::shared_mutex> s_lock (m_page_server_conn_vec_mtx);
   if (m_page_server_conn_vec.empty ())
     {
-      assert_release (false); // TODO some error-handling is needed such as shutdown
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_NO_PAGE_SERVER_CONNECTION, 0);
```

Comment:
The message for this error code reads:
```
At least one valid page server host is necessary, but no page server could be connected.
```
I think another one is needed.
Eg: `ER_NO_PAGE_SERVER_CONNECTION_REMAINING`.

Reply:
OK.